### PR TITLE
Right Click to Navigate History

### PR DESCRIPTION
**Pull request by dalhill**
## PR Checklist
- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change

## PR Type
- [x] Feature

## Fixes
Issue Number: #3474

## What is the current behavior?
In order to navigate history, the user must do so one step at a time.

## What is the new behavior?
User is able to right click the navigation buttons to move forward or backward multiple steps.

backward

forward

## Other information
Possible improvements
- [x] Limit the size of the history. Showing only 12 historical entries on either side. This is consistent with chrome.
- [x] Configure the list to dropdown the same height as the other header buttons.
- [x] If there is no "forward" or "backward" action to be taken, disable the button.

### CODE REVIEW

File: ui/component/router/view.jsx [hunk 1]
```diff
@@ -60,14 +62,50 @@ type Props = {
...
+    title,
   } = props;
+  const { channelName, streamName } = parseURI(uri);
+  const { entries } = history;
+  const entryIndex = history.index;
+
+  useEffect(() => {
+    if (typeof title !== 'undefined' && title !== '') {
+      document.title = title;
+    } else if (typeof streamName !== 'undefined' && streamName !== 'undefined' && streamName !== '') {
```
Comment (dalhill) [inline ui/component/router/view.jsx]: BTW - streamName was coming as a string of value 'undefined'. I didn't look too far into this bug.

File: ui/store.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ const persistOptions = {
 
 let history;
 // @if TARGET='app'
-history = createHashHistory();
+history = createMemoryHistory();
```
Comment (seanyesmunt) [inline ui/store.js]: What is the difference between these two?

File: ui/component/router/view.jsx [hunk 3]
```diff
@@ -60,14 +62,52 @@ type Props = {
...
     isAuthenticated,
+    history,
+    uri,
+    title,
   } = props;
+  const { channelName, streamName } = parseURI(uri);
+  const { entries } = history;
+  const entryIndex = history.index;
+
+  useEffect(() => {
```
Comment (seanyesmunt) [inline ui/component/router/view.jsx]: Why did you move this effect out of the show component?

File: ui/component/navigationButton/view.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          {entry.title}
+        </li>
+      );
+    },
+    [currentIndex, isBackward, go]
+  );
+  const slicedEntries = sliceEntries(currentIndex, entries, historyLength, isBackward);
+  return (
+    <div>
+      <Button
```
Comment (seanyesmunt) [inline ui/component/navigationButton/view.jsx]: Are you able to determine if we can go back here or not? If so, it would be great to disable the back button and give it some disabled style so it's easier to tell that you can't

**Comment by tzarebczan** [comment]
Thanks for the PR! Once this is fully reviewed/corrected/merged, please send us an email so we can show you some [appreciation](https://lbry.com/faq/appreciation)

File: ui/store.js (hunk 2 above)
Comment (dalhill) [reply ui/store.js]: createMemoryHistory gives us an array of entries so that we can reference the history stack.

*Excerpt from [history/docs/GettingStarted](https://github.com/ReactTraining/history/blob/master/docs/GettingStarted.md)*
> Additionally, createMemoryHistory provides history.index and history.entries properties that let you inspect the history stack.

File: ui/component/router/view.jsx (hunk 3 above)
Comment (dalhill) [reply ui/component/router/view.jsx]: I want this to fire on every page and not just the pages that render using the ShowPage component.

File: ui/component/navigationButton/view.jsx (hunk 4 above)
Comment (dalhill) [reply ui/component/navigationButton/view.jsx]: PR has been updated to disable button if no historical entries.

Comment (seanyesmunt) [reply ui/component/navigationButton/view.jsx]: Awesome!

**Comment by seanyesmunt** [comment]
Nice work @dalhill. This is really close.

I'll spend some time testing your branch this week just to make sure there aren't any issues on electron moving to memoryHistory, but I don't think there will be. I'll also push up a commit and do another pass at the styling.

**Comment by dalhill** [comment]
Pushed to resolve conflicts

File: ui/component/router/view.jsx (hunk 1 above)
Comment (seanyesmunt) [reply ui/component/router/view.jsx]: I think it's when you call `parseURI` with an `undefined` uri. Only calling `parseURI` if uri exists should fix it.
